tests/functional/check-refs.sh

```diff
@@ -60,3 +60,7 @@ if ! isTestOnNixOS; then
     fi
 
 fi
+
+# test12 should fail (syntactically invalid).
+expectStderr 1 nix-build -vvv -o "$RESULT" check-refs.nix -A test12 >"$TEST_ROOT/test12.stderr"
+grepQuiet -F "output check for 'lib' contains an illegal reference specifier 'dev', expected store path or output name (one of [lib, out])" < "$TEST_ROOT/test12.stderr"
```

Comment:
This assertion seems to fail

Reply:
The error message contains `error: derivation contains an illegal reference specifier 'dev'`, which looks like an error message from running daemon than newly built one.

Reply:
It must be when the tests are run with the daemon. You can gate this test on the daemon version, howeer.

Reply:
Are these tests ever run against the daemon built locally? Or is this test useless as a presubmit test and it will not catch a regression?

If I read your suggestion correctly I'm expected to use `isDaemonNewer`. How do I pick the version? Examples like `2.20.0pre20231220` are not very clear where the date comes from. Is it a base commit date? commit date? merge date?

Reply:
I think it's commit date. It seems like it wouldn't work, but in practice it appears to. I guess we don't test with daemons from earlier on the same day or whatever.

Reply:
Fixed as:

```diff
--- a/tests/functional/check-refs.sh
+++ b/tests/functional/check-refs.sh
@@ -63,2 +63,3 @@ fi

+if isDaemonNewer "2.28pre20241225"; then
     # test12 should fail (syntactically invalid).
@@ -66 +67,2 @@ expectStderr 1 nix-build -vvv -o "$RESULT" check-refs.nix -A test12 >"$TEST_ROOT
     grepQuiet -F "output check for 'lib' contains an illegal reference specifier 'dev', expected store path or output name (one of [lib, out])" < "$TEST_ROOT/test12.stderr"
+fi
```